csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImplSpecific.qll

```diff
@@ -308,3 +308,49 @@ ParameterPosition parseArgBody(string s) {
   s = "this" and
   result.isThisParameter()
 }
+
+/**
+ * Gets the valid model origin values.
+ */
+private string getValidModelOrigin() {
+  result =
+    [
+      "df", // Dataflow (model generator)
+      "tb", // Type based (model generator)
+    ]
+}
+
+/**
+ * A class used to represent provenance values for MaD models.
+ *
+ * The provenance value is a string of the form `origin-verification`
+ * (or just `manual`), where `origin` is a value indicating the
+ * origin of the model, and `verification` is a value indicating, how
+ * the model was verified.
+ *
+ * Examples could be:
+ * - `df-generated`: A model produced by the model generator, but not verified by a human.
+ * - `df-manual`: A model produced by the model generator, but verified by a human.
+ */
+class Provenance extends string {
+  private string verification;
+
+  Provenance() {
+    exists(string origin | origin = getValidModelOrigin() |
+      this = origin + "-" + verification and
+      verification = ["manual", "generated"]
+    )
+    or
+    this = verification and verification = "manual"
+  }
+
+  /**
+   * Holds if `this` is a valid generated provenance value.
```

Comment:
Again, should not use `` `this` `` in QL doc.

Reply:
That doesn't appear to be consistent around the codebase, but I will remove the backticks around `this`.